modules/videoio/src/cap_android_mediandk.cpp

```diff
@@ -257,6 +260,324 @@ class AndroidMediaNdkCapture : public IVideoCapture
+            LOGE("ERROR: findInfo");
+            return false;
+        }
+
+        format = AMediaFormat_new();
+        if (NULL == format) {
+            LOGE("ERROR: AMediaFormat_new");
+            goto error;
+        }
+
+        LOGV("mime: %s, width: %d, height: %d, fps: %f", info->mime, frameSize.width, frameSize.height, fps);
+
+        AMediaFormat_setString(format, AMEDIAFORMAT_KEY_MIME, info->mime);
+        AMediaFormat_setInt32(format, AMEDIAFORMAT_KEY_WIDTH, frameSize.width);
+        AMediaFormat_setInt32(format, AMEDIAFORMAT_KEY_HEIGHT, frameSize.height);
+        AMediaFormat_setFloat(format, AMEDIAFORMAT_KEY_FRAME_RATE, (float)fps);
+        AMediaFormat_setInt32(format, AMEDIAFORMAT_KEY_I_FRAME_INTERVAL, 5);
+        AMediaFormat_setInt32(format, AMEDIAFORMAT_KEY_BIT_RATE, frameSize.width * frameSize.height * 5);
+        AMediaFormat_setInt32(
+            format, AMEDIAFORMAT_KEY_COLOR_FORMAT,
+            #if __ANDROID_API__ >= 26
+            COLOR_FormatSurface
+            #else
+            COLOR_FormatYUV420SemiPlanar
+            #endif
+        );
+
+        encoder = AMediaCodec_createEncoderByType(info->mime);
+        if (NULL == encoder) {
+            LOGE("ERROR: AMediaCodec_createEncoderByType");
+            goto error;
+        }
+
+        status = AMediaCodec_configure(encoder, format, NULL, NULL, AMEDIACODEC_CONFIGURE_FLAG_ENCODE);
+        if (AMEDIA_OK != status) {
+            LOGE("ERROR: AMediaCodec_configure (%d)", status);
+            goto error;
+        }
+
+        #if __ANDROID_API__ >= 26
+        status = AMediaCodec_createInputSurface(encoder, &surface);
+        if (AMEDIA_OK != status || NULL == surface) {
+            LOGE("ERROR: AMediaCodec_createInputSurface (%d)", status);
+            goto error;
+        }
+        #endif
+
+        AMediaCodec_start(encoder);
+
+        fd = ::open(filename.c_str(), O_WRONLY | O_CREAT | O_TRUNC, 0666);
+        if (fd < 0) {
+            LOGE("ERROR: open");
+            goto error;
+        }
+
+        muxer = AMediaMuxer_new(fd, info->muxerFormat);
+        if (NULL == muxer) {
+            LOGE("ERROR: AMediaMuxer_new");
+            goto error;
+        }
+
+        AMediaMuxer_setOrientationHint(muxer, params.get(CAP_PROP_ORIENTATION_META, 0));
+
+        frameIndex = 0;
+        width = frameSize.width;
+        height = frameSize.height;
+        frameRate = fps;
+        videoTrackIndex = -1;
+
+        return true;
+
+    error:
+        close();
+        return false;
+    }
+
+    virtual void close()
+    {
+        if (videoTrackIndex >= 0 && NULL != muxer) {
+            drainEncoder(true);
+            AMediaMuxer_stop(muxer);
+        }
+
+        if (NULL != encoder) AMediaCodec_delete(encoder);
+        if (NULL != muxer) AMediaMuxer_delete(muxer);
+
+        #if __ANDROID_API__ >= 26
+        if (NULL != surface) ANativeWindow_release(surface);
+        #endif
+
+        if (fd >= 0) ::close(fd);
```

Comment:
According to open man page it returns -1 in case of error, else it's a valid value.
Maybe 0 will never be returned (I think is stdout) but I prefer to keep as: 
* < 0 - invalid
* >= valid

Reply:
ok